runtime/convertValues.go

```diff
@@ -172,15 +174,23 @@ func exportArrayValue(
 	cadence.Array,
 	error,
 ) {
-	elements := v.Elements()
-	values := make([]cadence.Value, len(elements))
+	values := make([]cadence.Value, 0, v.Count())
 
-	for i, value := range elements {
-		exportedValue, err := exportValueWithInterpreter(value, inter, seenReferences)
+	var err error
+	v.Iterate(func(value interpreter.Value) (resume bool) {
```

Comment:
There is no `Elements() []Value` function anymore, replace it with the new array iteration function